portal/flatpak-portal.c

```diff
@@ -1051,6 +1053,14 @@ check_for_updates (PortalFlatpakUpdateMonitor *monitor)
       return; /* Never report updates for uninstalled refs */
     }
 
+  dir = flatpak_installation_get_dir (installation, NULL);
+  if (dir == NULL)
+    return;
+
+  ref = flatpak_ref_format_ref (FLATPAK_REF (installed_ref));
+  if (flatpak_dir_ref_is_masked (dir, ref))
+    return; /* Never report updates for masked refs */
+
```

Comment:
Should we also check this in the code path that applies updates? Or does this cover both?

Reply:
That case is already handled by FlatpakTransaction